### PR TITLE
Allow setting the max iterations

### PR DESCRIPTION
**Pull request by wtdcode**
Trivial patch to allow users to set the max iterations for a `StdMutationalStage`

### CODE REVIEW

File: libafl/src/stages/mutational.rs [hunk 1]
```diff
@@ -238,7 +239,12 @@ where
 {
     /// Creates a new default mutational stage
     pub fn new(mutator: M) -> Self {
-        Self::transforming(mutator)
+        Self::transforming(mutator, DEFAULT_MUTATIONAL_MAX_ITERATIONS)
+    }
+
+    /// Creates a new mutational stage with the given max iterations
+    pub fn with_iterations(mutator: M, max_iterations: u64) -> Self {
```
Comment (domenukk) [inline libafl/src/stages/mutational.rs:246]: `with_max_iterations`?

**Comment by wtdcode** [comment]
> One of the fuzzers using it now errors:
> 
> ```rust
>     Checking baby_fuzzer_grimoire v0.10.1 (/Users/runner/work/LibAFL/LibAFL/fuzzers/baby_fuzzer_grimoire)
> error[E0061]: this function takes 2 arguments but 1 argument was supplied
>    --> src/main.rs:154:9
>     |
> 154 |         StdMutationalStage::transforming(grimoire_mutator)
>     |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^------------------ an argument of type `u64` is missing
> ```

Do you have some command to run these tests? I just checked `cargo test` and sent my PR.

**Comment by wtdcode** [comment]
Fixed!

**Comment by domenukk** [comment]
./scripts/test_all_fuzzers.sh - but probably it's faster to wait for CI :D

**Comment by domenukk** [comment]
Wait, is there any difference between `new` and `transforming`? Why do we have both constructors? Maybe we drop one?

**Comment by wtdcode** [comment]
> Wait, is there any difference between `new` and `transforming`? Why do we have both constructors? Maybe we drop one?

I have the same question when tweaking the code xd

**Comment by tokatoka** [comment]
> Wait, is there any difference between new and transforming? Why do we have both constructors? Maybe we drop one?
I have the same question when tweaking the code xd

the type constraints are different. it's needed for grimoire ig

**Comment by wtdcode** [comment]
Given the fact the `transforming` is also useful, I add a corresponding `transforming_with_max_iterations` for that.

**Comment by domenukk** [comment]
Thanks!

**Pull request merged**